src/Configuration.php

```diff
@@ -49,6 +49,22 @@ class Configuration
      **/
     private $xpath;
 
+     /**
+     * Boolean indicating if configuration results should be cached
+     *
+     * @access private
+     * @var    boolean
+     **/
+    private $useCaching;
+
+    /**
+     * The cache (used only when caching is enabled)
+     *
+     * @access private
+     * @var    array
+     **/
+    private $cache = ["alwaysArray" => [], "optionalArray" => [] ];
```

Comment:
The new array syntax took a little while to get used to for me, but after using it for a little while I really prefer it to the old syntax. I'd feel changing this actually reduces readability and would keep the software stuck in the old php structures (much like you would if you'd kept using the var statement and classnames for constructors in classes).